tests/drivers/uart/uart_elementary/testcase.yaml

```diff
@@ -0,0 +1,58 @@
+common:
+  tags: drivers uart
+  depends_on: gpio
+  harness: ztest
+  harness_config:
+    fixture: gpio_loopback
+tests:
+  drivers.uart.uart_elementary:
+    filter: CONFIG_SERIAL_SUPPORT_INTERRUPT
+    platform_allow:
+      - nrf54h20dk/nrf54h20/cpuapp
+      - nrf54l15pdk/nrf54l15/cpuapp
+      - nrf5340dk/nrf5340/cpuapp
+    extra_configs:
+      - CONFIG_UART_INTERRUPT_DRIVEN=y
+      - CONFIG_UART_0_INTERRUPT_DRIVEN=y
+      - CONFIG_UART_0_ENHANCED_POLL_OUT=n 
+      - CONFIG_UART_NRFX_UARTE_LEGACY_SHIM=n
+  drivers.uart.uart_elementary_dual_nrf54h:
+    filter: CONFIG_SERIAL_SUPPORT_INTERRUPT
+    platform_allow:
+      - nrf54h20dk/nrf54h20/cpuapp
+    extra_args: DTC_OVERLAY_FILE="boards/nrf54h20dk_nrf54h20_cpuapp_dual_uart.overlay"
+    extra_configs:
+      - CONFIG_UART_INTERRUPT_DRIVEN=y
+      - CONFIG_UART_0_INTERRUPT_DRIVEN=y
```

Comment:
configs for instance 0 and 1 can be removed. Is asynchronous  API is not enabled then enabling globally interrupt driven API will enable it for all instances.